### ISSUE TITLE
FATAL Flags parsing error: flag --use_gpu_relax=None

### ISSUE THREAD

**Issue by giangpth** [deepmind/alphafold#366]
Hello, 
I run the command 
`python3 docker/run_docker.py --output_dir=/home/t.pham1/Results/  --max_template_date=1980-05-14 --data_dir=/data/genetic_databases/ --fasta_paths=../Sequences/2C9V/rcsb_pdb_2C9V.fasta`
And got the error: 
```
I0214 22:18:45.409204 139904099993408 run_docker.py:87] Mounting /home/t.pham1/Sequences/2C9V -> /mnt/fasta_path_0
I0214 22:18:45.409291 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/uniref90 -> /mnt/uniref90_database_path
I0214 22:18:45.409355 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/mgnify -> /mnt/mgnify_database_path
I0214 22:18:45.409398 139904099993408 run_docker.py:87] Mounting /data -> /mnt/data_dir
I0214 22:18:45.409442 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/pdb_mmcif -> /mnt/template_mmcif_dir
I0214 22:18:45.409484 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/pdb_mmcif -> /mnt/obsolete_pdbs_path
I0214 22:18:45.409522 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/pdb70 -> /mnt/pdb70_database_path
I0214 22:18:45.409563 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/uniclust30/uniclust30_2018_08 -> /mnt/uniclust30_database_path
I0214 22:18:45.409606 139904099993408 run_docker.py:87] Mounting /data/genetic_databases/bfd -> /mnt/bfd_database_path
I0214 22:18:47.997892 139904099993408 run_docker.py:222] FATAL Flags parsing error: flag --use_gpu_relax=None: Flag --use_gpu_relax must have a value other than None.
I0214 22:18:47.998272 139904099993408 run_docker.py:222] Pass --helpshort or --helpfull to see help on flags.
```
I successfully run this command before but this time I got that error. 
Does anyone encounter the same problem?

**Comment by Augustin-Zidek** [comment]
Did you update to v2.1.2 recently? If so -- is it possible that you didn't update `docker/run_docker.py` when you did the update? That would explain why the `use_gpu_relax` flag is not being set. 

You can see the changes that should be present in the file in the diff for v2.1.2: https://github.com/deepmind/alphafold/commit/57a2455e5fab752d8df430ae1d0334e1a21b548c#diff-388496c8217a913453ccb72a5315f35b6ee463bd325f23ffd390b765dddbd477

**Comment by giangpth** [comment]
@Augustin-Zidek Thank you, it helps me solve that problem. But now i got another error
```
python3 docker/run_docker.py --output_dir=/home/t.pham1/Results/  --max_template_date=1980-05-14 --data_dir=/data/genetic_databases/ --fasta_paths=../Sequences/2C9V/rcsb_pdb_2C9V.fasta
I0215 14:29:58.523723 140551236130624 run_docker.py:107] Mounting /home/t.pham1/Sequences/2C9V -> /mnt/fasta_path_0
I0215 14:29:58.523831 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/uniref90 -> /mnt/uniref90_database_path
I0215 14:29:58.523884 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/mgnify -> /mnt/mgnify_database_path
I0215 14:29:58.523928 140551236130624 run_docker.py:107] Mounting /data -> /mnt/data_dir
I0215 14:29:58.523973 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/pdb_mmcif -> /mnt/template_mmcif_dir
I0215 14:29:58.524017 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/pdb_mmcif -> /mnt/obsolete_pdbs_path
I0215 14:29:58.524076 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/pdb70 -> /mnt/pdb70_database_path
I0215 14:29:58.524132 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/uniclust30/uniclust30_2018_08 -> /mnt/uniclust30_database_path
I0215 14:29:58.524178 140551236130624 run_docker.py:107] Mounting /data/genetic_databases/bfd -> /mnt/bfd_database_path
I0215 14:30:01.505030 140551236130624 run_docker.py:247] I0215 14:30:01.503818 140213830223680 templates.py:857] Using precomputed obsolete pdbs /mnt/obsolete_pdbs_path/obsolete.dat.
I0215 14:30:01.777196 140551236130624 run_docker.py:247] I0215 14:30:01.776188 140213830223680 tpu_client.py:54] Starting the local TPU driver.
I0215 14:30:01.777587 140551236130624 run_docker.py:247] I0215 14:30:01.776796 140213830223680 xla_bridge.py:212] Unable to initialize backend 'tpu_driver': Not found: Unable to find driver in registry given worker: local://
I0215 14:30:01.918856 140551236130624 run_docker.py:247] I0215 14:30:01.917777 140213830223680 xla_bridge.py:212] Unable to initialize backend 'tpu': Invalid argument: TpuPlatform is not available.
I0215 14:30:08.556120 140551236130624 run_docker.py:247] I0215 14:30:08.554941 140213830223680 run_alphafold.py:398] Have 5 models: ['model_1', 'model_2', 'model_3', 'model_4', 'model_5']
I0215 14:30:08.556490 140551236130624 run_docker.py:247] I0215 14:30:08.555121 140213830223680 run_alphafold.py:414] Using random seed 421031002483180752 for the data pipeline
I0215 14:30:08.556710 140551236130624 run_docker.py:247] I0215 14:30:08.555342 140213830223680 run_alphafold.py:163] Predicting rcsb_pdb_2C9V
I0215 14:30:08.556934 140551236130624 run_docker.py:247] I0215 14:30:08.556107 140213830223680 jackhmmer.py:133] Launching subprocess "/usr/bin/jackhmmer -o /dev/null -A /tmp/tmpaxt3kewt/output.sto --noali --F1 0.0005 --F2 5e-05 --F3 5e-07 --incE 0.0001 -E 0.0001 --cpu 8 -N 1 /mnt/fasta_path_0/rcsb_pdb_2C9V.fasta /mnt/uniref90_database_path/uniref90.fasta"
I0215 14:30:08.829277 140551236130624 run_docker.py:247] I0215 14:30:08.828245 140213830223680 utils.py:36] Started Jackhmmer (uniref90.fasta) query
I0215 14:35:06.288952 140551236130624 run_docker.py:247] I0215 14:35:06.287383 140213830223680 utils.py:40] Finished Jackhmmer (uniref90.fasta) query in 297.459 seconds
I0215 14:35:06.487032 140551236130624 run_docker.py:247] Traceback (most recent call last):
I0215 14:35:06.487389 140551236130624 run_docker.py:247] File "/app/alphafold/run_alphafold.py", line 445, in <module>
I0215 14:35:06.487612 140551236130624 run_docker.py:247] app.run(main)
I0215 14:35:06.487804 140551236130624 run_docker.py:247] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 312, in run
I0215 14:35:06.487988 140551236130624 run_docker.py:247] _run_main(main, args)
I0215 14:35:06.488203 140551236130624 run_docker.py:247] File "/opt/conda/lib/python3.7/site-packages/absl/app.py", line 258, in _run_main
I0215 14:35:06.488386 140551236130624 run_docker.py:247] sys.exit(main(argv))
I0215 14:35:06.488562 140551236130624 run_docker.py:247] File "/app/alphafold/run_alphafold.py", line 429, in main
I0215 14:35:06.488737 140551236130624 run_docker.py:247] is_prokaryote=is_prokaryote)
I0215 14:35:06.488909 140551236130624 run_docker.py:247] File "/app/alphafold/run_alphafold.py", line 177, in predict_structure
I0215 14:35:06.489081 140551236130624 run_docker.py:247] msa_output_dir=msa_output_dir)
I0215 14:35:06.489252 140551236130624 run_docker.py:247] File "/app/alphafold/alphafold/data/pipeline.py", line 174, in process
I0215 14:35:06.489454 140551236130624 run_docker.py:247] max_sto_sequences=self.uniref_max_hits)
I0215 14:35:06.489625 140551236130624 run_docker.py:247] File "/app/alphafold/alphafold/data/pipeline.py", line 102, in run_msa_tool
I0215 14:35:06.489796 140551236130624 run_docker.py:247] with open(msa_out_path, 'w') as f:
I0215 14:35:06.489967 140551236130624 run_docker.py:247] PermissionError: [Errno 13] Permission denied: '/mnt/output/rcsb_pdb_2C9V/msas/uniref90_hits.sto'
```
However, I am running through a remote machine without a full sudo permission so it may be the problem.

**Comment by Augustin-Zidek** [comment]
This is a permission issue on the database, could you `chmod` the `/mnt/output/` directory so that you have read/write/execute permissions on it?

**Comment by giangpth** [comment]
@Augustin-Zidek thank you so much, I will contact the admin of the remote machine for this as it is a permission stuff. Hence, I will close the issue.

**Comment by Yangyang-jane** [comment]
Hello,
I run the command
`bash run_alphafold.sh -d /share/pub/zhaohq/project/pumch/alphafold/version1/source/databases/ -o /share2/pub/yangjy/yangjy/softs/alphafold/result/query -f /share2/pub/yangjy/yangjy/softs/alphafold/example/query.fasta -t 2020-05-14`
And got the error:
`FATAL Flags parsing error: flag --use_gpu_relax=None: Flag --use_gpu_relax must have a value other than None.
Pass --helpshort or --helpfull to see help on flags.
`
I don't use Docker.
Does anyone encounter the same problem?

**Comment by Augustin-Zidek** [comment]
@Yangyang-jane You have to set the `--use_gpu_relax` flag to either `true` or `false`.

**Comment by Yangyang-jane** [comment]
> @Yangyang-jane You have to explicitly set the `--use_gpu_relax` flag to either `true` or `false`.
Thank you,I had solved this problem,but there existed another problem.I had set --use_gpu_relax flag True,but it still shows as the follow:
` UserWarning: Flag --use_gpu_relax has a non-None default value; therefore, mark_flag_as_required will pass even if flag is not specified in the command line!
  warnings.warn(
I0311 11:35:09.811530 140075759966016 templates.py:857] Using precomputed obsolete pdbs /share/pub/zhaohq/project/pumch/alphafold/version1/source/databases//pdb_mmcif/obsolete.dat.
I0311 11:35:10.234684 140075759966016 xla_bridge.py:247] Unable to initialize backend 'tpu_driver': NOT_FOUND: Unable to find driver in registry given worker: 
2022-03-11 11:35:11.258894: E external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
I0311 11:35:11.259597 140075759966016 xla_bridge.py:247] Unable to initialize backend 'gpu': FAILED_PRECONDITION: No visible GPU devices.
I0311 11:35:11.260360 140075759966016 xla_bridge.py:247] Unable to initialize backend 'tpu': INVALID_ARGUMENT: TpuPlatform is not available.
W0311 11:35:11.260569 140075759966016 xla_bridge.py:252] No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
I0311 11:35:15.515949 140075759966016 run_alphafold.py:397] Have 5 models: ['model_1', 'model_2', 'model_3', 'model_4', 'model_5']
I0311 11:35:15.516198 140075759966016 run_alphafold.py:414] Using random seed 1114105159714920706 for the data pipeline
I0311 11:35:15.516445 140075759966016 run_alphafold.py:163] Predicting query`
